fastlane_bot/helpers/txhelpers.py

```diff
@@ -168,10 +168,10 @@ def _create_transaction(self, contract, fn_name: str, args: list, value: int) ->
         }
 
     def _update_transaction(self, tx: dict):
-        tx["gas"] = self.cfg.w3.eth.estimate_gas(tx) # occasionally throws an exception
+        tx["gas"] = self.cfg.w3.eth.estimate_gas(tx) # may throw an exception
         if self.use_access_list:
-            result = self.cfg.w3.eth.create_access_list(tx) # rarely throws an exception
-            if tx["gas"] > result["gasUsed"]:
+            result = self.cfg.w3.eth.create_access_list(tx) # may return an error
+            if tx["gas"] > result["gasUsed"] and "error" not in result:
```

Comment:
I'd check the result is valid before accessing specific fields in it

Reply:
It is valid by the definition of function `create_access_list` - either it returns the `result` object, or it throws an exception (for example, if there is no network connection at that given point in time).